.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,20 @@
+name: Run All Tests
```

Comment:
caching dependencies and using the latest version of github actions might improve the workflow

```
on: [push]

jobs:
  run_examples:
    runs-on: ubuntu-20.04

    steps:
    - uses: actions/checkout@v4

    - name: Set up Python 3.9
      uses: actions/setup-python@v5
      with:
        python-version: 3.9

    - name: Cache Pip dependencies
      uses: actions/cache@v4
      with:
        path: ~/.cache/pip
        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
        restore-keys: |
          ${{ runner.os }}-pip-

    - name: Install Pip Dependencies
      run: pip install -r requirements.txt

    - name: Run Tests
      run: python -m pytest

    - name: List files
      run: ls

```

Reply:
> caching dependencies

I am not sure I'm understanding what this will do. Make pip run faster?

Reply:
This workflow currently installs all of the dependencies in requirements.txt every time that the workflow runs. You can cache them so that if the cache is found they do not need to be installed again in the workflow. Not a huge deal but saves time and processing power.

Reply:
Thanks, I will give it a try